src/api/users.js

```diff
@@ -1,362 +1,39 @@
-   * @param {string} userId id of the owner
-   * @returns {Object} object with the set of possible operations
-   */
-  docsOf (userId) {
-    return {
-      _owner: userId,
-      _blank: new ScopedServerCallBuilder().users(userId).documents(),
-      /**
-       * Get blob by id
-       * @param {string} docId
-       * @returns {Promise} A Promise with the response
-       */
-      get (docId, isSigned = false) {
-        return this._blank.id(docId).sign().get()
-      },
-
-      getUploadConfig (type, contentType) {
-        return this._blank
-          .sign()
-          .post({
-            data: {
-              type,
-              attributes: {
-                content_type: contentType
-              }
-            }
-          })
-      }
-    }
-  },
-
-  // legacy
-
-  getBlob ({ userId, blobId }) {
-    return new ScopedServerCallBuilder()
-      .users()
-      .id(userId)
-      .blobs()
-      .id(blobId)
-      .get()
-  },
-
-  getApproved () {
-    return server.sdkServer.users().approved().limit(store.getters.pageLimit).callWithSignature(store.getters.keypair)
-  },
-
-  getDisabled () {
-    return server.sdkServer.users().disabled().limit(store.getters.pageLimit).callWithSignature(store.getters.keypair)
-  },
-
-  getBalanceByAccountId (accountId, asset) {
-    return server.sdkServer.users()
-      .balance(accountId, asset).callWithSignature(store.getters.keypair)
-      .then(response => {
-        return response.records[0].balance_id
+      .then((response) => {
+        return response[address].email
       })
   },
 
-  getUnapproved () {
-    return server.sdkServer.users().regRequests().limit(store.getters.pageLimit).callWithSignature(store.getters.keypair)
-  },
-
-  getUserById (accountId) {
-    return server.sdkServer.users().accountId(accountId).callWithSignature(store.getters.keypair)
-  },
-
-  getUsersByStatus (status) {
-    return server.sdkServer.users().status(status).limit(store.getters.pageLimit).callWithSignature(store.getters.keypair)
-  },
-
-  getUserInfoById (accountId) {
-    return server.sdkServer.users().accountId(accountId).callWithSignature(store.getters.keypair)
-      .then(u => {
-        const email = u.email
-        const userType = u.user_type
-        let username = ''
-
-        switch (u.user_type) {
-          case 'individual': {
-            username = u.details.personal_details.first_name + ' ' + u.details.personal_details.last_name
-            break
-          }
-          case 'business': {
-            username = u.details.corporation_details.entity_name
-            break
-          }
-        }
-
-        if (username.length < 2) username = u.email
-
-        return { email, username, userType }
-      })
+  async getAccountIdByEmail (email) {
+    const response = await Sdk.api.users.getPage({ email: email })
+    const resultArray = response.data.filter(item => item.email === email)
+    return resultArray[0].id
   },
+  // legacy
 
   getUserIdByEmail (email) {
```

Comment:
we have the awesome method `getAccountIdByEmail` above, why do we still use this shit?

Reply:
bump

Reply:
not fixed wtf